common/archiver/archivalMetadata.go

```diff
@@ -73,25 +74,70 @@ const (
 
 // NewArchivalMetadata constructs a new ArchivalMetadata
 func NewArchivalMetadata(
+	dc *dynamicconfig.Collection,
 	historyStatus string,
 	historyReadEnabled bool,
 	visibilityStatus string,
 	visibilityReadEnabled bool,
 	domainDefaults *config.ArchivalDomainDefaults,
 ) ArchivalMetadata {
+	historyConfig := NewArchivalConfig(
+		historyStatus,
+		historyReadEnabled,
+		domainDefaults.History.Status,
+		domainDefaults.History.URI,
+	)
+	if historyConfig.ClusterConfiguredForArchival() {
+		// Only check dynamic config when archival is enabled in static config.
```

Comment:
Can you add to this comment a little. Basically summarize our one way config conversation.